libraries/ms-common/src/stm32f0xx/adc.c

```diff
@@ -0,0 +1,139 @@
+#include <stdbool.h>
+
+#include "adc.h"
+#include "log.h"
+#include "stm32f0xx.h"
+
+typedef struct ADCInterrupt {
+  ADCCallback callback;
+  void *context;
+  uint16_t reading;
+} ADCInterrupt;
+
+typedef struct ADCStatus {
+  ADCChannel current_channel;
+  bool continuous;
+} ADCStatus;
+
+static ADCInterrupt s_adc_interrupts[NUM_ADC_CHANNEL];
+static ADCStatus s_adc_status;
+
+static void prv_adc_conversion_sequence() {
```

Comment:
If you're only using it once and it isn't that complex, it probably doesn't need to be its own function.